frontend/src/queries/nodes/InsightQuery/utils/filtersToQueryNode.ts

```diff
@@ -104,6 +105,21 @@ export const cleanHiddenLegendSeries = (
               .map(([k]) => k)
         : undefined
 }
+export const sanitizeRetentionEntity = (entity: RetentionEntity | undefined): RetentionEntity | undefined => {
+    if (!entity) {
+        return undefined
+    }
+    const record = { ...entity }
+    for (const key of Object.keys(record)) {
+        if (!['id', 'kind', 'name', 'type', 'order', 'uuid', 'custom_name'].includes(key)) {
```

Comment:
If we were to add a new key to `RetentionEntity`, would we need to add it here too? May be worth a comment somewhere, otherwise we may get some confused devs in the future

Reply:
Perhaps... but I hope we won't need this soon anymore. As soon as the filters object is gone, this will be as well.

The backend has a more sensible allow-list approach though
https://github.com/PostHog/posthog/blob/4aa36d8d4f97a2b4403ee55c5b1d77a66087ed3f/posthog/hogql_queries/legacy_compatibility/filter_to_query.py#L145-L151

But since the tests are all green, I'll use the opportunity and merge it in. There will be more adjustments...